src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClient.cs

```diff
@@ -440,6 +501,74 @@ private void PublishContactPoints()
 
             _contactPathsPublished = _contactPaths;
         }
+
+        private void ProcessSubscribeMessage(Subscribe subscribe, IActorRef receptionist)
+        {
+            if (RegisterExternalSubscriber(subscribe))
+                receptionist.Tell(new PublishSubscribe.Subscribe(subscribe.Topic, Self, subscribe.Group));
+        }
+
+        private void ProcessUnsubscribeMessage(Unsubscribe unsubscribe, IActorRef receptionist)
+        {
+            if (UnregisterExternalSubscriber(unsubscribe))
+                receptionist.Tell(new PublishSubscribe.Unsubscribe(unsubscribe.Subscribe.Topic, Self, unsubscribe.Subscribe.Group));
+        }
+
+        private bool RegisterExternalSubscriber(Subscribe subscribe)
```

Comment:
1. I only meant to have local subscribers - every node needs to subscribe on its own. So no case of sending 20 messages instead of 2 that I've seen, but it is not explicit, I agree with you. 
I could use DistributedPubSub mediator and publish with preference to local actors, but I can't do that due to limitationI explained in previous comment - that I don't have original topic/group that  message was published to when I receive it.
2. Right. Need to address it somehow. What is the best way to do it? Use router or have some Watcher implementation?